packages/manager/apps/nasha-react/src/pages/index.tsx

```diff
@@ -1,53 +1,48 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { Link, Navigate, Outlet } from 'react-router-dom';
-import {
-  QueryClient,
-  QueryClientProvider,
-  useQuery,
-} from '@tanstack/react-query';
-import { services } from '../api/nasha-react';
+import { Navigate, Outlet } from 'react-router-dom';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { getServices } from '../api/nasha-react';
+import Datagrid from '@/components/layout-helpers/list/dataGrid';
 
 const queryClient = new QueryClient();
 
-function Services() {
-  const { isLoading, isError, data } = useQuery(['listNasha'], services);
+function ServiceList({ data }) {
   const { t } = useTranslation('nasha-react');
 
-  if (isLoading) {
-    return <span>Loading...</span>;
-  }
-
-  if (isError) {
-    return <span>Error...</span>;
-  }
-
   const count = data?.length;
-  if (count === 0) {
-    return <Navigate to="onboarding" />;
-  }
+  if (count === 0) return <Navigate to="onboarding" />;
   if (count === 1) {
     return (
       <>
         <Outlet />
-        <Navigate to={data[0]} />
+        <Navigate to={data[0].serviceName} />
       </>
     );
   }
   return (
     <>
       <h2>{t('title')}</h2>
-      <ul>
-        {data.map((serviceName: string) => (
-          <li key={serviceName}>
-            <Link to={`/details/${serviceName}`}>{serviceName}</Link>
-          </li>
-        ))}
-      </ul>
+
+      <Datagrid data={data} />
     </>
   );
 }
 
+function Services() {
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
```

Comment:
We use ReactQuery or Suspense to be able to display a loading state and error state usually.
And do we have a spec that says we want to navigate to the onboarding page if there is a problem on the listing page ?

Reply:
That's right, we use ReactQuery, my bad. Fixed ✅